lua/render-markdown/handler/markdown.lua

```diff
@@ -222,6 +222,30 @@ M.render_code = function(buf, info)
         },
     }
     list.add(marks, background_mark)
+
+    if code.width == 'block' then
+        -- overwrite anything beyond block width + right_pad with Normal
+        local width = 0
+        local lines = vim.api.nvim_buf_get_lines(buf, start_row, end_row, false)
+        for _, line in ipairs(lines) do
+            width = math.max(width, str.width(line))
+        end
+        local pad = str.pad('', vim.o.columns * 2 - width)
+        for row = start_row, code.border == 'thin' and end_row or end_row - 1 do
+            ---@type render.md.Mark
+            local block_background_mark = {
+                conceal = false,
+                start_row = row,
+                start_col = 0,
+                opts = {
+                    virt_text = { { pad, 'Normal' } },
+                    virt_text_win_col = width + code.right_pad,
```

Comment:
This also needs to account for left padding:

```lua
virt_text_win_col = code.left_pad + width + code.right_pad,
```

Reply:
fixed!